Fix missing result in _discrete_log_trial_mul

joprodrigues opened the pull request:
#### References to other Issues or PRs
Fixes second part of #27612


#### Brief description of what is fixed or changed
There was a off-by-one error when _discrete_log_trial_mul, which meant that if the solution was equal to the order, it would not be found

#### Other comments


#### Release Notes

<!-- BEGIN RELEASE NOTES -->

* ntheory
  * Fixed missed solutions in discrete_log, when solution is equal to order

<!-- END RELEASE NOTES -->

joprodrigues committed:
Fix missing result in _discrete_log_trial_mul

There was a off-by-one error when _discrete_log_trial_mul is called
with an order

haru-44 commented:
I do not intend to merge this PR because I am assuming this [kind of code](https://github.com/sympy/sympy/issues/27612#issuecomment-2700778796).

joprodrigues committed:
Fix discrete_log for (large) composite n

_discrete_log_pohlig_hellman does not support composite n when b is not
a generator.
Handle the composite n case explicitly

joprodrigues commented:
Hi,

Could you give me some feedback regarding the newest changes, @haru-44 ?
Should I take into account https://github.com/sympy/sympy/pull/27948, and remove the order and prime_order arguments (since currently they are almost superfluous)?
